examples/platformer/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+)
+
+const (
+	// Settings
+	width           int  = 1024
+	height          int  = 512
+	fullscreen      bool = false
+	runinbackground bool = true
+)
+
+var (
+	LoadedSprite    *ebiten.Image
+	LeftSprite      *ebiten.Image
+	RightSprite     *ebiten.Image
+	IdleSprite      *ebiten.Image
+	BackgroundImage *ebiten.Image
+
+	IsFirstFrame bool = true
+
+	BackgroundOptions = &ebiten.DrawImageOptions{}
+	CharSpriteOptions = &ebiten.DrawImageOptions{}
+)
+
+func update(screen *ebiten.Image) error {
+	// Draws Background Image
+	screen.DrawImage(BackgroundImage, BackgroundOptions)
+
+	// Resets
+	charX := 0.0
+	charY := 0.0
+
+	// I'm doing this to easily define a starting position for the gopher
+	if IsFirstFrame == true {
+		charY = charY + 390
+		charX = charX + 50
+		IsFirstFrame = false
+	}
+
+	// Controls
+	if ebiten.IsKeyPressed(ebiten.KeyA) {
+		// Selects preloaded sprite
+		LoadedSprite = LeftSprite
+		// Moves character 3px right
+		charX = charX - 3.0
+	} else if ebiten.IsKeyPressed(ebiten.KeyD) {
+		// Selects preloaded sprite
+		LoadedSprite = RightSprite
+		// Moves character 3px left
+		charX = charX + 3.0
+	} else {
+		LoadedSprite = IdleSprite
+	}
+
+	// Change gopher's position
+	CharSpriteOptions.GeoM.Translate(charX, charY)
+
+	// FPS counter
+	fps := fmt.Sprintf("FPS: %f", ebiten.CurrentFPS())
+	ebitenutil.DebugPrint(screen, fps)
+
+	// Draws selected sprite image
+	screen.DrawImage(LoadedSprite, CharSpriteOptions)
+
+	return nil
+}
+
+func main() {
+	// Settings for images
+	BackgroundOptions.GeoM.Scale(0.5, 0.5)
+	BackgroundOptions.GeoM.Translate(0, 0)
+	CharSpriteOptions.GeoM.Scale(0.5, 0.5)
+
+	// Preload images, This isn't proper error handling however has been kept short for simplicity purposes
+	RightSprite, _, _ = ebitenutil.NewImageFromFile("img/right.png", ebiten.FilterNearest)
+	LeftSprite, _, _ = ebitenutil.NewImageFromFile("img/left.png", ebiten.FilterNearest)
+	IdleSprite, _, _ = ebitenutil.NewImageFromFile("img/mainchar.png", ebiten.FilterNearest)
+	BackgroundImage, _, _ = ebitenutil.NewImageFromFile("img/background.png", ebiten.FilterNearest)
```

Comment:
1. The image paths should be in `ebiten/examples/_resources/images` for the convention of other examples.
2. What are the images' license? Are they compatible with Ebiten's license (Apache License 2.0)?
3. Don't ignore the error here since IO error might still happen. Just panicing is fine.

Reply:
Can't get the image pathing to work.

rightSprite, _, err = ebitenutil.NewImageFromFile("_resources/images/platformer/right.png", ebiten.FilterNearest)

panic: open _resources/images/platformer/right.png: no such file or directory

Am i just missing something obvious?

Reply:
Nevermind, I forgot I have to use the example tag when running the script....